pkg/sql/create_table.go

```diff
@@ -63,6 +64,13 @@ func (p *planner) CreateTable(ctx context.Context, n *tree.CreateTable) (planNod
 	var sourcePlan planNode
 	var synthRowID bool
 	if n.As() {
+		if containsWindowFunc(p.txCtx, n.AsSource) {
```

Comment:
Is there a way to use this mechanism:

```go
                        scalarProps := &p.semaCtx.Properties
                        defer scalarProps.Restore(*scalarProps)
                        scalarProps.Require("FILTER", tree.RejectSpecial)
```

or alternatively (see `sort.go`)

```go
        if p.semaCtx.Properties.Derived.SeenGenerator {
                return nil, tree.NewInvalidFunctionUsageError(tree.GeneratorClass, "ORDER BY")
        }
```

Reply:
It's OK, thank you for checking. All this code is going away in 19.2 anyway so we shouldn't spend time over-thinking this.